### PR TITLE
Removing limit of 100 jobs analyzed by adding pagination to pipelines

### PR DESCRIPTION
**Pull request by EricTendian**
Thanks for making this script! I've found it really useful. I happen to have quite a lot of builds, so the 100 build limit wasn't going to cut it. I looked into the GitLab API and it looks like an easy fix is just make multiple calls to the pipelines API, each with a different page. This means we don't need to limit it to 100 jobs anymore.

I've tested this on limits under 100, multiples of 100, and numbers between multiples of 100 (e.g. 156, 232, etc) so I feel confident the pagination is working properly.

(P.S. - I would recommend ignoring whitespace changes when reviewing the diff)

### CODE REVIEW

File: gitlab_jobs.py
Comment (mgedmin) [inline gitlab_jobs.py]: I think you meant to do `per_page=per_page` here.

File: gitlab_jobs.py
Comment (mgedmin) [inline gitlab_jobs.py]: The diff (and also the code itself, I think) can be made more readable if you extract this bit into a helper function.  Something like

```python
def get_pipelines(args):
    ...
    for page in ...
        ...
        for pipeline in project.pipelines.list(...):
            yield pipeline
```

and then in main() you'd replace the `pipelines = ...` assignment to call the new helper and then you can keep the same `for pipeline in pipelines:` loop at the same indentation.

File: gitlab_jobs.py
Comment (mgedmin) [inline gitlab_jobs.py]: I have an unreasonable fear of floating point so I'd do it with integers:

```python
    pages = (args.limit + max_per_page - 1) // max_per_page
```

File: gitlab_jobs.py
Comment (mgedmin) [inline gitlab_jobs.py]: flake8 is going to insist that I add spaces around the `+`.

**Review by mgedmin — CHANGES_REQUESTED** [review]
Thank you very much for this PR!  It looks excellent overall, but I have made some suggestions and I'll leave it up to you to decide if you want to apply them or not before I merge this.

Only one of my suggestions is important: the per_page argument passed to project.pipelines.list() seems wrong. I've no idea how you get 101 items instead of 200 if you ask for two pages with per_page=101 set on both.

**Comment by EricTendian** [comment]
@mgedmin thanks for the great feedback! I haven't used the `yield` keyword much before, but this was a great use case. Also, not sure how I missed the `per_page=args.limit`, I had a different script to test things out so might've been a copy-paste error.

Anyway, the changes have been implemented!

**Review by mgedmin — APPROVED** [review]
Thank you very much!

**Pull request merged**
